lightning/src/ln/outbound_payment.rs

```diff
@@ -824,14 +824,15 @@ impl OutboundPayments {
 		}
 	}
 
-	fn push_path_failed_evs_and_scids<I: ExactSizeIterator + Iterator<Item = Result<(), APIError>>>(
+	fn push_path_failed_evs_and_scids<I: ExactSizeIterator + Iterator<Item = Result<(), APIError>>, L: Deref>(
 		payment_id: PaymentId, payment_hash: PaymentHash, route_params: &mut RouteParameters,
-		paths: Vec<Vec<RouteHop>>, path_results: I, pending_events: &Mutex<Vec<events::Event>>
-	) {
+		paths: Vec<Vec<RouteHop>>, path_results: I, logger: &L, pending_events: &Mutex<Vec<events::Event>>
+	) where L::Target: Logger {
 		let mut events = pending_events.lock().unwrap();
 		debug_assert_eq!(paths.len(), path_results.len());
 		for (path, path_res) in paths.into_iter().zip(path_results) {
 			if let Err(e) = path_res {
+				log_error!(logger, "Failed to send along path due to error: {:?}", e);
 				if let APIError::MonitorUpdateInProgress = e { continue }
```

Comment:
Not a big deal, but ideally we shouldn't `log_error` if we get `APIError::MonitorUpdateInProgress`, because it's not an error (🙄). Could conditionally log or move the current log to after the `continue`

Reply:
ah thanks